### PR TITLE
v1.9 backports 2020-10-21

### PR DESCRIPTION
**Pull request by nathanjsweet**
v1.9 backports 2020-10-21

 * [x] #13630 -- contrib: match commit subject exactly when searching for upstream commit (@tklauser)
 * [ ] #13376 -- ci: run baseline perf tests nightly (@nebril)
 * [ ] #13639 -- images: Fix handing of dev suffix when tag is used (cilium/image-tools#76) (@errordeveloper)
 * [x] #13604 -- k8s/watchers: fix data race in (*K8sWatcher).addK8sServiceV1 (@tklauser)
 * [ ] #13299 -- Add log when allocate nodecidr failure (@konghui)
 * [x] #13295 -- test: Debug RuntimeConntrackInVethModeTest flake (@pchaigno)
 * [x] #13612 -- eni: Fix data race during multiple updates at once (@christarazi)
 * [ ] #13640 -- docs: Document some caveats of kube-proxy replacement (@brb)
 * [ ] #13638 -- bpf: fix up lrp for v4-in-v6 sockets (@borkmann)
 * [ ] #13424 -- Expose operator azure-user-assigned-identity-id flag to its chart (@ombre9)
 * [ ] #13632 -- docs/gettingstarted: Update AKS instructions (@twpayne)
 * [ ] #13646 -- bpf: redirect fixes and follow-ups (@borkmann)
 * [x] #13643 -- docker: update Hubble CLI to v0.7.0 (@Rolinh)
 * [x] #13642 -- checkpatch: switch to an external container image (@qmonnet)
 * [ ] #13620 -- vendor: pin yaml.v2 to v2.2.8 (@twpayne)
 * [ ] #13550 -- operator: Fix CEP owner type (@jrajahalme)
 * [x] #13651 -- docs: fix minor issue in cilium support with external etcd gsg (@fristonio)
 * [x] #13644 -- Fixes for troubleshooting guide re. Hubble/Hubble Relay (@tklauser)
 * [ ] #13645 -- docs: GKE - fix some indentation, specify bash code segments (@ti-mo)
 * [x] #13665 -- docs: NodePort XDP on GCP is not supported (@gandro)
 * [x] #13654 -- k8s: update k8s libraries to 1.19.3 (@aanm)
 * [x] #13661 -- docs: Fix broken formating and link (@pchaigno)
 * [x] #13607 -- helm: Remove hardcoded port check for hubble, etc (@nathanjsweet)
 * [x] #13677 -- helm: keep encryption interface value undefined (@kkourt)
 * [x] #13684 -- install/kubernetes: remove nodePort.device from values (@tklauser)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 13630 13376 13639 13604 13299 13295 13612 13640 13638 13424 13632 13646 13643 13642 13620 13550 13651 13644 13645 13665 13654 13661 13607 13677 13684; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
LGTM for my commits

**Review by pchaigno — APPROVED** [review]
My changes look good :+1:

**Review by kkourt — APPROVED** [review]

**Review by rolinh — APPROVED** [review]

**Review by brb — APPROVED** [review]
LGTM for my changes.

**Review by qmonnet — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
👍 for my changes

**Comment by errordeveloper** [comment]
My commit looks good!

**Comment by gandro** [comment]
test-backport-1.9

**Review by fristonio — APPROVED** [review]
LGTM for my commit.

**Comment by nathanjsweet** [comment]
test-backport-1.9

**Comment by nathanjsweet** [comment]
**tldr;** checkpatch is to be ignored

`make -C bpf checkpatch` is being overly pedantic about the commit message. I tried fixing it once and was able to reduce the error to a warning, but the checkpatch script still exits > 0 even on warnings. I decided to go back to the original commit message structure, because it seemed important. All the other bpf checks are successful, and checkpatch is **only** complaining about the commit messages not the code in anyway.

**Comment by pchaigno** [comment]
> **tldr;** checkpatch is to be ignored

@nathanjsweet I noticed the same and added one error I think would be worth ignoring at https://github.com/cilium/cilium/issues/12467.

**Review by joestringer — APPROVED** [review]
I did a brief look overall at the change. It looks big, but there's net -9K lines from #13642 and a further +5K -1K from #13620 so that covers the vast majority of the changes. That seems sensible.

I asked the backporter and all commits were cleanly cherry-picked. plus realistically v1.9 is so close to master at this point that we're unlikely to introduce breakage just by making v1.9 closer to master via these backports. We've had several authors check over the backports as well so most of it should be right. I think it's better to get these backports in and get them regularly running against CI and rely on the next RC to provide further evaluation of the implementation.

If you as an author notice that one of the PRs is being backported and might need extra attention, by all means go ahead and review the backport here & post feedback so we can make sure the backport is done right. We can always iterate on top if further changes are necessary.
